### PR TITLE
Set IGNORE value in mask to -torch.inf

### LINKED ISSUE #318: [Bug Report] Sometimes attention score on real text is < - 1e5 , set attn.IGNORE to -infinity
Minimal example:

```
model = HookedTransformer.from_pretrained("pythia-70m")
t2 = torch.tensor([   0, 1814, 1149, 1538, 1063, 1753, 1870, 1873, 1058, 1686, 1249, 1435,
        1134, 1557, 1230, 1845, 1459, 1188, 1135, 1219, 1545, 1284, 1278, 1810,
        1858, 1486, 1801, 1939, 1697, 1612, 1014, 1987, 1892, 1872, 1541, 1137,
        1386, 1703, 1355, 1197, 1228, 1331, 1394, 1383, 1261, 1629, 1559, 1309,
        1342, 1615, 1425, 1562, 1158, 1624, 1823, 1459, 1389, 1734, 1252, 1212,
        1422, 1666, 1792, 1940, 1599, 1685, 1516, 1116, 1946, 1929, 1347, 1373,
        1442, 1982, 1416, 1248, 1820, 1439, 1300, 1452, 1114, 1971, 1140, 1384,
        1593, 1922, 1382, 1206, 1290, 1333, 1298, 1733, 1897, 1880, 1585, 1234,
        1868, 1921, 1133, 1670, 1847, 1002, 1861, 1661, 1690, 1346, 1025, 1652,
        1245, 1256, 1221, 1388, 1455, 1785, 1297, 1299, 1166, 1583, 1027, 1108,
        1791, 1083, 1959, 1250, 1519, 1270, 1063, 1999, 1059, 1814, 1149, 1538,
        1063, 1753, 1870, 1873, 1058, 1686, 1249, 1435, 1134, 1557, 1230, 1845,
        1459, 1188, 1135, 1219, 1545, 1284, 1278, 1810, 1858, 1486, 1801, 1939,
        1697, 1612, 1014, 1987, 1892, 1872, 1541, 1137, 1386, 1703, 1355, 1197,
        1228, 1331, 1394, 1383, 1261, 1629, 1559, 1309, 1342, 1615, 1425, 1562,
        1158, 1624, 1823, 1459, 1389, 1734, 1252, 1212, 1422, 1666, 1792, 1940,
        1599, 1685, 1516, 1116, 1946, 1929, 1347, 1373, 1442, 1982, 1416, 1248,
        1820, 1439, 1300, 1452, 1114, 1971, 1140, 1384, 1593, 1922, 1382, 1206,
        1290, 1333, 1298, 1733, 1897, 1880, 1585, 1234, 1868, 1921, 1133, 1670,
        1847, 1002, 1861, 1661, 1690, 1346, 1025, 1652, 1245, 1256, 1221, 1388,
        1455, 1785, 1297, 1299, 1166, 1583, 1027, 1108, 1791, 1083, 1959, 1250,
        1519, 1270, 1063, 1999, 1059])
_, c2 = model.run_with_cache(t2)
line(c2["attn_scores", 4][0, 3, 214])
```

This gives attention scores that are -116,000 on the correct segment, such that the pattern is uniform across things *outside* the causal mask. I have no idea why this happens, and it seems ridiculous, but we should set attn.IGNORE to -torch.inf probably

### PR DESCRIPTION
**Pull request by neelnanda-io**
# Description

Fixes #318 - In pythia-70m, attn_scores can get below -1e5 (for some reason...). Masked attn scores are set to -1e5, so this makes the model attend outside the mask! This is dumb, and so I set things to -torch.inf instead. It's still registered as a buffer so should move to device + dtypes accordingly

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

### Screenshots

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### CODE REVIEW

**Comment by ArthurConmy** [comment]
@neelnanda-io are you looking for someone to add the tests and docs? Or a review on this? CC @curt-tigges and others blocked

**Review by ArthurConmy — CHANGES_REQUESTED** [review]
Please also make this update to `HookedEncoder` (used for BERT):

https://github.com/neelnanda-io/TransformerLens/blob/c268a7159a6f8d5c78236a3f958f2d704fbc940f/transformer_lens/HookedEncoder.py#L123

**Comment by neelnanda-io** [comment]
Change made

File: transformer_lens/HookedEncoder.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def forward(
 
         resid = self.hook_full_embed(self.embed(tokens, token_type_ids))
 
-        large_negative_number = -1e5
+        large_negative_number = -torch.inf
         additive_attention_mask = (
             large_negative_number
             * repeat(1 - one_zero_attention_mask, "batch pos -> batch 1 1 pos")
```
[the remark this answers is not in the document]
Comment (ArthurConmy) [reply transformer_lens/HookedEncoder.py:126]: You need to replace the additive attention_mask definition with this one --- sorry didn't realize this first time!

**Comment by jbloomAus** [comment]
@ArthurConmy Maybe we can get someone who isn't Neel to finish this? Maybe an ARENA student?

**Comment by ArthurConmy** [comment]
Superseded by https://github.com/neelnanda-io/TransformerLens/pull/366
